networkx/drawing/layout.py

```diff
@@ -1121,6 +1122,235 @@ def multipartite_layout(G, subset_key="subset", align="vertical", scale=1, cente
     return pos
 
 
+def forceatlas2_layout(
+    G,
+    pos=None,
+    *,
+    n_iter=100,
+    jitter_tolerance=1.0,
+    scaling_ratio=2.0,
+    gravity=1.0,
+    distributed_action=False,
+    strong_gravity=False,
+    adjust_sizes=False,
+    mass="mass",
+    size="size",
+    dissuade_hubs=False,
+    weight=None,
+    linlog=False,
+    dim=2,
+):
+    """Forceatlas2 layout.
+
+    The ForceAtlas2 layout is a force-directed approach that improves upon
+    ForceAtlas by simplifying the parameters [1]_. It offers a good compromise
+    between clustering and spatially separating parts of  the network.
+    The layout was originally designed and used by Gephi to visualize
+    graphs in a continuous and interactive manner.
+
+    In the layout, nodes repulse each other similar to oppositely charged particles,
+    while edges attract like springs. The balance between the attractive and
+    repulsive forces produce visually pleasing layouts, particularly for large graphs.
+
+    The algorithm includes parameters for customization, such as gravity (pull
+    towards the center), scaling (size of the layout) and jitter (random perturbations).
+
+    Parameters
+    ----------
+    G : nx.Graph
+       Networkx graph
+    pos: dict or None
+       Optional starting positions
+    n_iter: number (default: 100)
+        Simulation steps
+    jitter_tolerance : number (default: 1.0)
+        Jitter tolerance for adjusting speed of layout convergence; lower values
+        are slower but yield more accuracy. Jitter controls how much swinging is
+        allowed. Values above 1 are discouraged. Jitter helps break symmetries
+        between similar parts of the graph through random perturbations in the
+        layout generation.
+    scaling_ratio : number (default: 2.0)
+        Controls force scaling constants k_attraction and k_repulsion
+    distributed_action : bool (default: False)
+        Normalizes the attraction forces according to the degree of the nodes
+    strong_gravity : bool (default: false)
+        Controls the "pull" to the center of mass of the plot
+    mass: str (default: "mass")
+        Looks up node attribute mass. The mass increases the attractive pull towards other nodes, similar
+    to a planetary body. The value defaults to 1.
+    size: str (default: "size")
+        Looks up node attribute size. .Allows for node sizes to have a larger size, preventing a "crowding effect".
```

Comment:
This suggestion tweaks the doc_strings for the new parameters. But I am not convinced that we've got the right defaults here. If you really want to go the route of storing these values on the graph object, can we have the default value be `None` just like we do for `weight`? That will reduce the chance of a user unknowingly setting the size or mass just because they have a different attribute named "mass" or "size". This would also allow us to eliminate the `adjust_sizes` parameter since we could check whether `size is None`.
```
    mass: str (default: None)
        Name of a node attribute indicating the node mass. The mass increases the attractive
        pull towards other nodes, similar to a planetary body. 
        If the attribute does not exist the default mass is ``1 + G.degree[node]``.
    size: str (default: None)
        Name of a node attribute indicating the node size. Size prevents a "crowding effect"
        if nodes get closer that the indicated size.
        If None, or the node attribute does not exist for a node, the default size is 1.
```

Reply:
How about we change the mass and size to a dict specifying node to mass/size, and replacing the strings?

Reply:
That sounds reasonable. We could even allow string or a dict. But that's getting more tricky.

Do these values affect the layout strongly. Are the layouts you show above in this PR just using the defaults? What is driving you to include these parameters as opposed to just using the default values?

Reply:
In the images above are generated with the defaults. In general the scaling and gravity have strong effects on the layout. This means that the mass potentially has a large effect. The size parameter helps prevent crowding. They are nice to have in there to give the full implementation of the layout as presented in the original publication. 

Does this suggestion work for you too @rossbar?

Reply:
I agree with @dschult  - I'd vote for the defaults of both the `size` and `mass` kwargs to be `None`.

Reply:
OK -- let's stick with strings for these parameters and None as the default value (meaning use the default for all nodes).